frontend/ui/elements/menu_keyboard_layout.lua

```diff
@@ -1,17 +1,67 @@
 local CheckButton = require("ui/widget/checkbutton")
+local CheckMark = require("ui/widget/checkmark")
+local Device = require("device")
 local FFIUtil = require("ffi/util")
+local Font = require("ui/font")
 local Language = require("ui/language")
+local Size = require("ui/size")
+local TextWidget = require("ui/widget/textwidget")
 local UIManager = require("ui/uimanager")
 local VirtualKeyboard = require("ui/widget/virtualkeyboard")
+local Screen = Device.screen
+local T = require("ffi/util").template
 local dbg = require("dbg")
+local logger = require("logger")
 local util = require("util")
 local _ = require("gettext")
 
 local input_dialog, check_button_bold, check_button_border, check_button_compact
 
+local function getActivatedKeyboards(compact)
+    local keyboard_layouts = G_reader_settings:readSetting("keyboard_layouts", {})
+    local activated_keyboards = {}
+    for lang, dummy in FFIUtil.orderedPairs(VirtualKeyboard.lang_to_keyboard_layout) do
+        if util.arrayContains(keyboard_layouts, lang) then
+            if compact then
+                table.insert(activated_keyboards, lang)
+            else
+                table.insert(activated_keyboards, Language:getLanguageName(lang))
+            end
+        end
+    end
+    if #activated_keyboards == 0 then
+        logger.dbg("menu_keyboard_layout: use default keyboard")
+        if compact then
+            return VirtualKeyboard:getKeyboardLayout(), 1
+        else
+            return Language:getLanguageName(VirtualKeyboard:getKeyboardLayout()), 1
+        end
+    end
+    return table.concat(activated_keyboards, ", "), #activated_keyboards
+end
+
 local sub_item_table = {
     {
-        text = _("Keyboard layout"),
+        text_func = function()
+            local activated_keyboards, nb_keyboards = getActivatedKeyboards()
+            local item_text = string.format(_("Keyboard layouts: %s"), activated_keyboards)
```

Comment:
Maybe `T` too?

Reply:
Just changed that  You were 3 sec faster ;)